zookeeper-server/src/main/java/org/apache/zookeeper/server/quorum/Learner.java

```diff
@@ -682,4 +683,14 @@ public void shutdown() {
     boolean isRunning() {
         return self.isRunning() && zk.isRunning();
     }
+
+    void closeSocket() {
+        try {
+            if (sock != null && !sock.isClosed()) {
```

Comment:
Just curious: why checking 'isClosed' ? Can't we call  'close'  multiple times?

Reply:
You're completely right, this check doesn't actually contribute anything - I'll remove it.